tools/automation/cli_linter/linter.py

```diff
@@ -0,0 +1,154 @@
+
+    def get_commands(self):
+        return self._commands
+
+    def get_command_groups(self):
+        return self._command_groups
+
+    def get_help_file_entries(self):
+        return self._help_file_entries
+
+    def get_command_parameters(self, command_name):
+        return self._parameters.get(command_name)
+
+    def get_help_entry_type(self, entry_name):
+        return self._all_yaml_help.get(entry_name).get('type')
+
+    def get_command_help(self, command_name):
+        return self._get_loaded_help_description(command_name)
+
+    def get_command_group_help(self, command_group_name):
+        return self._get_loaded_help_description(command_group_name)
+
+    def get_parameter_help(self, command_name, parameter_name):
+        options = self._command_table.get(command_name).arguments.get(parameter_name).type.settings.get('options_list')
+        parameter_helps = self._loaded_help.get(command_name).parameters
+        param_help = next((param for param in parameter_helps if _share_element(options, param.name.split())), None)
+        # workaround for --ids which is not does not generate doc help (BUG)
+        if not param_help:
+            return self._command_table.get(command_name).arguments.get(parameter_name).type.settings.get('help')
+        return param_help.short_summary or param_help.long_summary
+
+    def mark_rule_failure(self):
+        self._exit_code = 1
+
+    def get_exit_code(self):
+        return self._exit_code
+
+
+    def _get_loaded_help_description(self, entry):
+        return self._loaded_help.get(entry).short_summary or self._loaded_help.get(entry).long_summary
+
+
+class LinterManager():
+    def __init__(self, command_table=None, help_file_entries=None, loaded_help=None):
+        self.linter = Linter(command_table=command_table, help_file_entries=help_file_entries, loaded_help=loaded_help)
+        self._exclusions = None
+
+        self._rules = {
+            'help_file_entries': [],
+            'command_groups': [],
+            'commands': [],
+            'params': []
+        }
+
+    def get_help_entry_exclusions(self):
+        return self._exclusions.get('help_entries')
+
+    def get_command_group_exclusions(self):
+        return self._exclusions.get('command_groups')
+
+    def get_command_exclusions(self):
+        return self._exclusions.get('commands')
+
+    def get_parameter_exclusions(self, command_name):
+        return self._exclusions.get('params').get(command_name, {})
+
+    def add_rule(self, rule_type, rule_callable):
+        if rule_type in self._rules:
+            self._rules.get(rule_type).append(rule_callable)
+
+    def run(self, run_params=None, run_commands=None, run_command_groups=None, run_help_files_entries=None):
+        paths = import_module('automation.cli_linter.rules').__path__
+        exclusion_path = os.path.join(paths[0], 'exclusions.json')
+        self._exclusions = json.load(open(exclusion_path))
```

Comment:
If there is no exclusions file, does this still work?

Reply:
Definitely not, I can make it so it does.